torch/distributed/_tensor/device_mesh.py

```diff
@@ -432,10 +432,9 @@ def all_reduce(
                 to reduce on.
 
         Returns:
-            A :class:`Work` object
+            A :class:`torch.Tensor` object
```

Comment:
would this be AsyncTensor or just Tensor?

Reply:
It will be AsyncTensor in eager and plain tensor when tracing. Right @wconstab ?

Either way, do we want to document this fact?